docs/architecture/adr-065-store-v2.md

```diff
@@ -165,58 +164,60 @@ otherwise, a snapshot could be triggered on a height that is not available in SC
 The state sync process should be largely unaffected by the separation of the SC
 and SS layers. However, if a node syncs via state sync, the SS layer of the node
 will not have the state synced height available, since the IAVL import process is
-not setup in way to easily allow direct key/value insertion. A modification of
-the IAVL import process would be necessary to facilitate having the state sync
-height available.
+not setup in way to easily allow direct key/value insertion.
 
-Note, this is not problematic for the state machine itself because when a query
-is made, the RMS will automatically direct the query correctly (see [Queries](#queries)).
+We propose a simple `SnapshotManager` that consumes and produces snapshots, that
+syncs to both the SC and SC backends.
 
-#### Queries
+#### RootStore
 
-To consolidate the query routing between both the SC and SS layers, we propose to
-have a notion of a "query router" that is constructed in the RMS. This query router
-will be supplied to each `KVStore` implementation. The query router will route
-queries to either the SC layer or the SS layer based on a few parameters. If
-`prove: true`, then the query must be routed to the SC layer. Otherwise, if the
-query height is available in the SS layer, the query will be served from the SS
-layer. Otherwise, we fall back on the SC layer.
+We will define a `RootStore` interface and default implementation that will be
+the primary interface for the application to interact with. The `RootStore` will
+be responsible for housing SS and SC backends. Specifically, a `RootStore` will
+provide the following functionality:
 
-If no height is provided, the SS layer will assume the latest height. The SS
-layer will store a reverse index to lookup `LatestVersion -> timestamp(version)`
-which is set on `Commit`.
+* Manage commitment of state (both SS and SC)
+* Provide modules access to state
+* Query delegation (i.e. get a value for a <key, height> tuple)
+* Providing commitment proofs
+
+#### Store Keys
+
+Naturally, if a single SC tree is used in all RootStore implementations, then the
+notion of a store key becomes entirely useless. However, we cannot dictate or
+predicate how all applications will implement their RooStore (if they choose to).
+
+Since an app can choose to have multiple SC trees, we need to keep the notion of
+store keys. Unlike store v1, we represent store keys as simple strings as opposed
+to concrete types to provide OCAP functionality. The store key strings act to
```

Comment:
should we add this as a consequence to document the changes around ocap more clearly

Reply:
Good idea. Updated.